xDrip/Scenes/Settings/BaseSettings/BaseSettingsModels.swift

```diff
@@ -14,12 +14,12 @@ enum BaseSettings {
         case textInput(mainText: String, detailText: String?, textChangedHandler: (String?) -> Void)
         case rightSwitch(text: String, isSwitchOn: Bool, switchHandler: (Bool) -> Void)
         case volumeSlider(value: Float, changeHandler: (Float) -> Void)
-        case pickerExpandable(mainText: String, detailText: String?, picker: UIPickerView)
+        case pickerExpandable(mainText: String, detailText: String?, dataSource: [[String]], picker: UIView)
     }
     
     enum Section {
         case normal(cells: [Cell], header: String?, footer: String?)
-        case singleSelection(cells: [String], header: String?, footer: String?, selectionHandler: (Int) -> Void)
+        case singleSelection(cells: [(String, Bool)], header: String?, footer: String?, selectionHandler: (Int) -> Void)
```

Comment:
`singleSelection` means you may have only one selected cell in the section. Than what's the purpose of `(String, Bool)` tuple? Leave it be `[String]` and add an additional argument - `selectedIndex: Int`.